src/libraries/System.Net.Ping/src/System/Net/NetworkInformation/Ping.RawSocket.cs

```diff
@@ -103,19 +106,16 @@ private static Socket GetRawSocket(SocketConfig socketConfig)
             {
                 // If it is not multicast, use Connect to scope responses only to the target address.
                 socket.Connect(socketConfig.EndPoint);
-                unsafe
+                ReadOnlySpan<byte> opt = BitConverter.IsLittleEndian ? [1, 0, 0, 0] : [0, 0, 0, 1];
```

Comment:
FWIW, I get that this removes a use of unsafe and we're all about that, but the new code is much less intuitive and harder to understand, imo

Reply:
@stephentoub an alternative would be
```cs
int opt = 1;
MemoryMarshal.AsBytes(new ReadOnlySpan<int>(in opt))
```
which likely will be marked as unsafe anyway

Reply:
@jkotas @stephentoub so what's the preferred action here? leaving unsafe for readability?

Reply:
We could certainly consider more strongly-typed overloads; some such overloads exist for SetSocketOption. We didn't because a) SetRawSocketOptions is already an escape hatch that's a last resort when you need to use a non-cross-platform value, and b) the underlying APIs themselves just take a byte sequence, so this is exposing the raw capability how it's exposed from the OS. I'd leave it to @karelz and co as to whether they want to add additional overloads here.

In the meantime, my preference is the MemoryMarshal. You can even do it without a separate local, as the compiler will create one for you, but IIRC you'll get a warning diagnostic.
```C#
MemoryMarshal.AsBytes(new ReadOnlySpan<int>(1))
```